file/builder.go

```diff
@@ -748,6 +746,32 @@ func (b *stateBuilder) plugins() {
 	}
 }
 
+// strip_path schema default value is 'true', but it cannot be set when
+// protocols include 'grpc' and/or 'grpcs'. When users explicitly set
+// strip_path to 'true' with grpc/s protocols, deck returns a schema violation error.
+// When strip_path is not set and protocols include grpc/s, deck sets strip_path to 'false',
+// despite its default value would be 'true' under normal circumstances.
+func getStripPathBasedOnProtocols(initialStripPath *bool, route kong.Route) (*bool, error) {
+	var grpcProtocols bool
+	for _, tag := range route.Protocols {
+		if *tag == "grpc" || *tag == "grpcs" {
+			grpcProtocols = true
+		}
+	}
+
+	if grpcProtocols {
+		if initialStripPath != nil {
+			if *initialStripPath {
+				return nil, fmt.Errorf("schema violation (strip_path: cannot set " +
+					"'strip_path' when 'protocols' is 'grpc' or 'grpcs')")
+			}
+			return initialStripPath, nil
+		}
+		return kong.Bool(false), nil
+	}
```

Comment:
Question: If strip path can't be set when protocol is grpc, why return false on the two returns above?

Reply:
`strip_path` is a required field, so when protocol is grpc it needs to be set to `false` (`true` being the schema default)

Reply:
```go
	for _, p := range route.Protocols {
		if *p == "grpc" || *p == "grpcs" {
			if stripPath != nil { return nil, fmt.Errorf("strip path not allowed")
      return kong.Bool(false), nil
		}
	}
```
Is that not sufficient?

Reply:
Not directly...I just reworked/cleaned a bit the code to have something like you are suggesting (https://github.com/Kong/deck/pull/573/commits/67d1dcc9485e7754195449dff24f1a663ae4f4eb)